src/components/interaction/tabs/AvTabs/AvTabs.test.ts

```diff
@@ -47,70 +90,191 @@ BddTest().given('a tab switcher ', () => {
+        beforeEach(async () => {
+          await wrapper.findAll('.av-tab-item__tab')[0].trigger('keydown', { key: 'ArrowLeft' })
+        })
+
+        BddTest().then('it should emit an update to the last tab', async () => {
+          expect(wrapper.emitted('update:modelValue')?.at(-1)?.[0]).toBe(2)
+        })
+      })
+
+      BddTest().and('the home key is pressed', () => {
+        beforeEach(async () => {
+          await wrapper.findAll('.av-tab-item__tab')[0].trigger('keydown', { key: 'Home' })
+        })
+
+        BddTest().then('it should not emit an update', async () => {
+          expect(wrapper.emitted('update:modelValue')).toBeUndefined()
+        })
       })
-    })
 
-    BddTest().when('selecting the previous tab on the first tab', () => {
-      BddTest().then('it should update the active tab to the last tab', async () => {
-        await wrapper.findAllComponents({ name: 'DsfrTabItem' })[0].vm.$emit('previous')
-        expect(wrapper.emitted('update:modelValue')?.at(-1)?.[0]).toBe(2)
+      BddTest().and('the end key is pressed', () => {
+        beforeEach(async () => {
+          await wrapper.findAll('.av-tab-item__tab')[0].trigger('keydown', { key: 'End' })
+        })
+
+        BddTest().then('it should emit an update to the last tab', async () => {
+          expect(wrapper.emitted('update:modelValue')?.at(-1)?.[0]).toBe(2)
+        })
       })
     })
 
-    BddTest().when('selecting the first tab', () => {
+    BddTest().when('the second tab is active', () => {
       beforeEach(() => {
         wrapper = mount(AvTabs, { props: { modelValue: 1 }, slots, global: { stubs } })
       })
 
-      BddTest().then('it should set the active tab to index 0', async () => {
-        await wrapper.findAllComponents({ name: 'DsfrTabItem' })[1].vm.$emit('first')
-        expect(wrapper.emitted('update:modelValue')?.at(-1)?.[0]).toBe(0)
+      BddTest().and('the right arrow key is pressed', () => {
+        beforeEach(async () => {
+          await wrapper.findAll('.av-tab-item__tab')[1].trigger('keydown', { key: 'ArrowRight' })
+        })
+
+        BddTest().then('it should emit an update to the third tab', async () => {
+          expect(wrapper.emitted('update:modelValue')?.at(-1)?.[0]).toBe(2)
+        })
       })
-    })
 
-    BddTest().when('selecting the last tab', () => {
-      BddTest().then('it should set the active tab to the last index', async () => {
-        await wrapper.findAllComponents({ name: 'DsfrTabItem' })[0].vm.$emit('last')
-        expect(wrapper.emitted('update:modelValue')?.at(-1)?.[0]).toBe(2)
+      BddTest().and('the left arrow key is pressed', () => {
+        beforeEach(async () => {
+          await wrapper.findAll('.av-tab-item__tab')[1].trigger('keydown', { key: 'ArrowLeft' })
+        })
+
+        BddTest().then('it should emit an update to the first tab', async () => {
+          expect(wrapper.emitted('update:modelValue')?.at(-1)?.[0]).toBe(0)
```

Comment:
Crée une function getLastEmittedEvent par exemple et l'utiliser partout dans le fichier
```suggestion
          expect(wrapper.emitted('update:modelValue')?.at(-1)?.[0]).toBe(0)
```